### PR TITLE
[MAPPS-2572, MAPPS-2573] Splunk TA Bug fixes

### PR DESCRIPTION
**Pull request by munna-shaik-s1**
Jira Link: <https://sentinelone.atlassian.net/browse/MAPPS-2572>, <https://sentinelone.atlassian.net/browse/MAPPS-2573>

# 🥅 Goal

The reported Bug fixes

# 🛠️ Solution

Provide description of changes

# 🏫 Testing

How the changes were tested

### CODE REVIEW

**Comment by alexandrmaximov-s1** [comment]
A few questions:
- do we support 1/0 values for `tenant`? There is not mentioned in the specification. It should rather be changed to checkbox.
- there is a mismatch in first part of the token and the second one, `authn_token_part_two` doesn't have `"encrypted": true`  flag, probably it need to have it as well. 
  - iirc one of these two parts wasn't laoded/stored correctly when a user tries to edit changes

File: TA_dataset/bin/dataset_common.py [hunk 1]
```diff
@@ -170,8 +170,11 @@ def get_acct_info(self, logger, account=None):
                     acct_dict[conf.name]["ds_api_key"] = get_token_from_config(
                         self, conf, conf.name, logger
                     )
-                    if hasattr(conf, "tenant"):
-                        tenant_value = get_tenant_value(conf, logger)
+                    if hasattr(conf, "tenant") and conf.tenant in [
+                        "all_scopes",
+                        "specified_scopes",
+                    ]:
+                        tenant_value = True if conf.tenant == "all_scopes" else False
```
Comment (alexandrmaximov-s1) [inline TA_dataset/bin/dataset_common.py:177]: This code repeats 3x. Can we extract it to a separate function?

File: TA_dataset/bin/dataset_common.py [hunk 2]
```diff
@@ -233,13 +217,13 @@ def get_acct_info(self, logger, account=None):
...
 
 
-def get_tenant_value(conf, logger):
-    tenant_value = conf.tenant
-    tenant_value = tenant_value.strip()
-    logger.debug("The provided tenant value in config is {}".format(tenant_value))
-    if tenant_value.lower() == "false" or tenant_value.lower() == "0":
-        return False
-    return True
+def get_updated_account_dict(conf, conf_name, acct_dict, logger):
```
Comment (alexandrmaximov-s1) [inline TA_dataset/bin/dataset_common.py:220]: Can we have a more descriptive name, e.g. `update_tenant_conf`

Comment (munna-shaik-s1) [reply TA_dataset/bin/dataset_common.py]: Done

**Review by alexandrmaximov-s1 — APPROVED** [review]

**Pull request merged**